guides/admin/inotify-watch-limits.md

```diff
@@ -0,0 +1,266 @@
+---
+title: Troubleshooting inotify Watcher Limit Problems
+description: Learn how to resolve problems related to the inotify Watcher Limit.
+---
+
+The [`inotify` facility] allows programs to monitor files and directories for
+changes, so that they will receive an event immediately whenever a user or
+program modifies the file or directory. Many such programs can fall back to
+periodically checking files and directories for changes, also known as polling.
+Using polling avoids the watch limit, but may result in higher system loading
+and a longer interval for the program to detect changes.
+
+`inotify` requires kernel resources (memory and processor) for each file it
+tracks. As a result, the Linux kernel limits the number of file watchers that
+each user can register. The default settings vary according to the host system
+distribution; on Ubuntu 20.04 LTS, the default limit is 8,192 watches per instance
+
+[`inotify` facility]: https://en.wikipedia.org/wiki/Inotify
+
+With some applications and tools, including Webpack or [code-server], you may
+encounter an error similar to the following:
+
+> Watchpack Error (watcher): Error: ENOSPC: System limit for number of file
+> watchers reached, watch '/some/path'
+
+[code-server]: https://github.com/cdr/code-server
+
+On a 64-bit system, each `inotify` watch that programs register will consume
+approximately 1 kB of kernel memory, which cannot be swapped to disk and is not
+counted against the environment memory limit setting.
+
+## Diagnosis
+
+If the total number of watchers is close to the `max_user_watches` setting, then
+it is likely that you are encountering this limit.
+
+### Check tunable settings
+
+There are three kernel tuning options related to the `inotify` system:
+
+- `fs.inotify.max_queued_events`, which defines an upper bound on the number of
+  file notification events pending delivery to programs.
+- `fs.inotify.max_user_instances`, which determines the maximum number of
+  `inotify` instances per user. Programs using `inotify` will typically create
+  a single _instance_, so this limit is unlikely to cause issues.
+- `fs.inotify.max_user_watches`, which defines the maximum number of files and
+  folders that programs can monitor for changes.
```

Comment:
Suggestion: use `:` followed be `defines` to reduce verbosity as shown below

```md
- `fs.some_option`: defines ...
```

Reply:
good suggestion, thanks!